### PR TITLE
Feature/interval type

### PR DESCRIPTION
**Pull request by geosp**
Number search refactoring.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR has lint errors. Please fix these and commit them.
      
<details>
  <summary>number.js</summary>
  <p>Full Path: <code>/home/circleci/project/src/example-types/number.js</code></p>
  
  <strong>Expected method shorthand.</strong>
  <code>Line 23: result: async (context, search) => {</code>

  <strong>Expected property shorthand.</strong>
  <code>Line 46: interval: interval,</code>

  <strong>'totalDocuments' is assigned a value but never used.</strong>
  <code>Line 53: let totalDocuments = _.get('hits.total', valuesResult)</code>

  <strong>Unexpected block statement surrounding arrow body; parenthesize the returned value and move it immediately after the `=>`.</strong>
  <code>Line 54: histogram = _.map(entry => {</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The changelog has not been updated. Please update the changelog.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The version was not updated. Please update the version.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  We were able to automatically fix some formatting issues in this PR for you!
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/example-types/number.js [hunk 1]
```diff
@@ -20,4 +20,53 @@ module.exports = {
...
+            field,
+            missing: 0,
+          },
+        },
+      },
+    })
+
+    let statistical = _.get('aggregations.statistical', result)
+    let interval =
+      Math.round(Math.abs(statistical.max - statistical.min) / 50) || 1
```
Comment (daedalus28) [inline src/example-types/number.js]: Do we want 50 buckets?

File: src/example-types/number.js [hunk 2]
```diff
@@ -20,4 +20,53 @@ module.exports = {
...
+            histogram: {
+              field,
+              interval,
+              min_doc_count: 0,
+            },
+          },
+        },
+      })
+
+      let totalDocuments = _.get('hits.total', valuesResult)
```
Comment (daedalus28) [inline src/example-types/number.js]: lint error was right - this variable isn't used

File: src/example-types/number.js [hunk 3]
```diff
@@ -20,4 +20,53 @@ module.exports = {
       }),
     },
   }),
+  async result(context, search) {
+    let field = _.get('field', context)
```
Comment (daedalus28) [inline src/example-types/number.js]: probably better to just destructure in the args

**Pull request merged**